### PR TITLE
feat: support button in extension setting

### PR DESCRIPTION
**Pull request by igoogolx**

### CODE REVIEW

File: app/renderer/ui/preference-view/components/extension-setting.vue [hunk 1]
```diff
@@ -15,9 +16,19 @@ const props = defineProps({
...
     required: true,
   },
+  name: { type: String, required: true },
 });
 
 const emits = defineEmits(["event:change"]);
+
+const onButtonClick = () => {
+  PLAPI.logService.info(
+    props.name,
```
Comment (GeoffreyChen777) [inline app/renderer/ui/preference-view/components/extension-setting.vue:26]: why do we need this `name`?

and the log should be removed?

File: app/renderer/ui/preference-view/components/button.vue [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    type: String,
+    required: true,
+  },
+});
+
+const emits = defineEmits(["event:change"]);
+</script>
+
+<template>
+  <div class="flex justify-between">
```
Comment (GeoffreyChen777) [inline app/renderer/ui/preference-view/components/button.vue:17]: could you please provide a screenshot of this component?

I wish to check the UI looking. thanks.

File: app/renderer/ui/preference-view/components/extension-setting.vue (hunk 1 above)
Comment (igoogolx) [reply app/renderer/ui/preference-view/components/extension-setting.vue:26]: - Users need feedback when the button is clicked, which is why the log should be kept. 
- The `name` tells where the log is from.

image

File: app/renderer/ui/preference-view/components/button.vue (hunk 2 above)
Comment (igoogolx) [reply app/renderer/ui/preference-view/components/button.vue:17]: image

File: app/renderer/ui/preference-view/components/extension-setting.vue (hunk 1 above)
Comment (GeoffreyChen777) [reply app/renderer/ui/preference-view/components/extension-setting.vue:26]: I think we don't need to trigger the log here.

We can do it in the extension code.

The extension developer listens to this button in the extension code, once user click it, run corresponding function, such as clearing the cache, and how a log that the cache has been cleared.

File: app/renderer/ui/preference-view/components/button.vue (hunk 2 above)
Comment (GeoffreyChen777) [reply app/renderer/ui/preference-view/components/button.vue:17]: UI looks ok.

File: app/renderer/ui/preference-view/components/extension-setting.vue (hunk 1 above)
Comment (igoogolx) [reply app/renderer/ui/preference-view/components/extension-setting.vue]: In [7211188](https://github.com/Future-Scholars/paperlib/pull/606/commits/7211188ac41673c5f208e8ff4c09637bce259950) and [bbe480e](https://github.com/Future-Scholars/paperlib-ai-chat-extension/commit/bbe480ee360ddd7f15e6d2ce2eb04ac189a26e3a), the extension will listen to the change of preference value. Now the log can be removed.

**Review by GeoffreyChen777 — APPROVED** [review]

**Pull request merged**